request-authentication.md

```diff
@@ -6,13 +6,85 @@ pagination_next: null
 pagination_prev: null
 ---
 
+import ApiSchema from '@theme/ApiSchema';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
 # Request authentication
 
 A notification request from webhooks can be verified by using the `host`, `x-ms-date`, `x-ms-content-sha256`, and `authorization` headers with the
 unique secret received when the webhook was registered. The authorization is a
 sha256 HMAC hash of the request date, content, and URI that is created using the
 webhook secret.
 
+<Tabs
+    defaultValue="js" 
+    groupId="example-choice" 
+    values={[ {label: 'JavaScript', value: 'js'}, {label: '.Net C#', value: 'csharp'}, ]}>
+
+<TabItem value="js">
+
+To run: `node webhook.mjs`
+```js
+import crypto from "crypto";
+
+// The secret is found in the response body when you register a webhook. 
+// Note: The secret is unique per hook registration (EventType+YourURL).
+const secret = 'A0+AeKBRG2KRGvnNwJpQlb6IJFk48CKXCIcrLoHncVJKDILsQSxS6NWCccwWm6r6FhGKhiHTBsG2wo/xU6FY/A=='
+
+// The complete url of the webhook, including query parameters if applicable. I.e. https://yoursite.com/webhooks/vmp/v1
+const requestUrl = "https://webhook.site/e2cee29b-012e-4f1d-8ef4-e95fd74a7a63"
+// Part of the HTTP request header from Vipps MobilePay, found in the field 'x-ms-date'
+const dateTime = "Thu, 30 Mar 2023 08:38:32 GMT"
+
+// The signature/authorization sent by Vipps MobilePay for you to verify against. 
+// Part of the HTTP request header from Vipps MobilePay, found in the field 'authorization'
+const actualSignature = "HMAC-SHA256 SignedHeaders=x-ms-date;host;x-ms-content-sha256&Signature=agAiSyogQbDHpeucoNwYz+yAr5nJ+v+zasdkSbqzv+U="
+
+// HTTP Body
+const payloadJson = {
+  "some-unique-content":"ee6e441b-cc4a-46f8-895d-a5af79bcc233/hello-world"
+}
+
+const hashedPayload =  crypto.createHash("sha256").update(JSON.stringify(payloadJson)).digest("base64")
+// returns 'lNlsp1XA03N34HrQsVzPgJKtC+r7l/RBF4V3JQUWMj4='
+
+// The complete url of the call to the webhook, including query parameters if applicable
+const url = new URL(requestUrl)
+const pathAndQuery = url.pathname + url.search; 
+// returns '/e2cee29b-012e-4f1d-8ef4-e95fd74a7a63'
+const host = url.host 
+// returns 'webhook.site'
```

Comment:
Suggestion, keep comments on the right side or above for blocks,
it's confusing to read if it's below too.

Also, I'm confused by 'returns', nothing returns here, it's 'evaluates too',
but the common format I know is just `const host = url.host // webhook.site`

Reply:
Fixed for most cases